Language Bindings (#2)
servicebinding/website: README.md

The specification mandates the `type` field and recommends `provider` field in the projected binding.  In many cases the `type` field should be good enough to select the appropriate binding.  In cases where it is not (e.g. when there are different providers for the same Provisioned Service), the `provider` field may be used.  For example, when the type is `mysql`, the `provider` value might be `mariadb`, `oracle`, `bitnami`, `aws-rds`, etc.  When the workload is selecting a binding, if necessary, it could consider `type` and `provider` as a composite key to avoid ambiguity.  This could be helpful if a workload needs to choose a particular provider based on the deployment environment.  In the deployment environment (`stage`, `prod`, etc.), at any given time, you need to ensure only one binding projection exists for a given `type` or `type` and `provider` -- unless your workload needs to connect to all the services.

### Environment Variables

> The specification also has support for projecting binding values as environment variables.  You can use the built-in language feature of your programming language of choice to read environment variables.  The container must restart to update the values of environment variables if there is a change in the binding.

Reviewer: Distinguish between a container restart due to a binding change vs needing to restart to pick up an updated value

```suggestion
The specification also has support for projecting binding values as environment variables.  You can use the built-in language feature of your programming language of choice to read environment variables.  The container must restart to read updated environment variable values.
```

        Connection conn;
        try {
            conn = DriverManager.getConnection(url);
        } catch (SQLException e) {
            System.err.printf("Unable to connect to database: %s", e);
            System.exit(1);
        }

        // ...
    }

}
```

### Go
_(Example taken from <https://github.com/baijum/servicebinding>)_

```go
import (
	"context"
	"fmt"
	"github.com/jackc/pgx/v4"
	"github.com/baijum/servicebinding/binding"
	"os"
)

func main() {
	sb, err := bindings.FromServiceBindingRoot()
	if err != nil {
		_, _ = fmt.Fprintln(os.Stderr, "Could not read service bindings")
		os.Exit(1)
	}
	
	b, err := sb.Bindings("postgresql")
	if err != nil {
		_, _ = fmt.Fprintln(os.Stderr, "Unable to find postgresql binding")
		os.Exit(1)
	}
	if len(b) != 1 {
		_, _ = fmt.Fprintf(os.Stderr, "Incorrect number of PostgreSQL bindings: %d\n", len(b))
		os.Exit(1)
	}

	u, ok := b[0]["url"]
	if !ok {
		_, _ = fmt.Fprintln(os.Stderr, "No URL in binding")
		os.Exit(1)
	}

	conn, err := pgx.Connect(context.Background(), u)
	if err != nil {
		_, _ = fmt.Fprintf(os.Stderr, "Unable to connect to database: %v\n", err)
		os.Exit(1)
	}
	defer conn.Close(context.Background())

	// ...
}
```

> ## Known Langage Bindings

Reviewer 2: typo
```suggestion
## Known Language Bindings
```
